### PR TITLE
added header defaults

### PR DESCRIPTION
**Pull request by johnhof**
@NickMele added the default we talked about. setting a `reqHeaders` property to `undefined` will remove it. Added a unit test for it

### CODE REVIEW

**Comment by johnhof** [comment]
integration tests are still failing, but this is a step in the right direction

**Comment by johnhof** [comment]
https://github.com/johnhof/express-proxy/issues/8

File: lib/proxy.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ var _       = require('lodash');
 var URL     = require('url');
 var async   = require('async');
 var colors  = require('colors');
+var helpers = require('./helpers');
```
Comment (NickMele) [inline lib/proxy.js:6]: classic :wink:

Comment (NickMele) [inline lib/proxy.js:6]: although i think you forgot to include this file, i am not seeing it

**Comment by NickMele** [comment]
Didn't you say that github requires certain headers? Would that make the first 2 integration tests fail?

**Comment by johnhof** [comment]
updated with renamed files/helpers file. as for the failing tests, you are correct. I think its ok for now. I'm just trying to get this headers logic in. I'm thinking we should get unit tests for the config working and then move on to integration testing. Thats more of the last step (I also dont have the bandwidth to debug it at the moment, haha)

**Comment by johnhof** [comment]
@NickMele  any issues here or can I merge?

**Comment by NickMele** [comment]
Have not had a chance to test it

Nick

File: lib/helpers.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+var _ = require('lodash')
+
+exports.defaultHeaders = function (overrideObj, defaultObj) {
+  var result = defaultObj ? _.clone(defaultObj, true) : {};
```
Comment (NickMele) [inline lib/helpers.js]: can headers contain objects? is it necessary to do the deep clone?

**Comment by NickMele** [comment]
@johnhof overall everything looks pretty good.

my one recommendation would be that since this is going to be a public module, and i guess even if it wasn't. you might want to add a little more error handling into the helper rather than assuming everything passed into it will be what you are expecting. like checking everything is an object and probably that each header value is a string.

**Comment by johnhof** [comment]
@NickMele  I see what you're saying, I just dotn want to presuppose anything on the user. if they really want the `Content-Type` to be an object, so be it.  I'll add some more fallbacks/sanity checks though.

File: lib/helpers.js (hunk 2 above)
Comment (johnhof) [inline lib/helpers.js]: see comment below

**Comment by NickMele** [comment]
I guess, but i mean there is a difference between supporting a standard and allowing the user to just do what they want.

So if the object is supposed to contain headers, and headers can only be defined as strings we know it will never work right?

**Comment by johnhof** [comment]
@NickMele I'm not 100% sold on this but I dont think its worth  disputing it. updated

**Comment by johnhof** [comment]
@NickMele bump

**Pull request merged**